### ISSUE TITLE
No `libgit2-dev` package found on `apt` (Ubuntu 20.04.4 LTS)

### ISSUE THREAD

**Issue by bit-man** [kdeldycke/meta-package-manager#579]
Fails to install `libgit2-dev` package using apt even that the package exists

```
$ mpm install libgit2-dev
warning: Skip unavailable apm manager.
error:   WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
error: 
error:   E: Invalid operation version
warning: Skip unavailable apt-mint manager.
warning: Skip unavailable brew manager.
warning: Skip unavailable composer manager.
warning: Skip unavailable dnf manager.
warning: Skip unavailable flatpak manager.
warning: Skip unavailable gem manager.
warning: Skip unavailable npm manager.
warning: Skip unavailable opkg manager.
warning: Skip unavailable pacman manager.
warning: Skip unavailable vscode manager.
warning: Skip unavailable yarn manager.
Package manager order: 
warning:   WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
warning: No libgit2-dev package found on apt.
warning: No way to search for libgit2-dev with pip. Try to directly install it.
Install libgit2-dev package from pip...
error:   ERROR: Could not find a version that satisfies the requirement libgit2-dev (from versions: none)
error:   ERROR: No matching distribution found for libgit2-dev
warning: Could not install libgit2-dev with pip.
error:   error: unknown flag `color'
warning: No libgit2-dev package found on snap.
```

Doing a search with apt the package is found 
```
$ apt search libgit2-dev
Sorting... Done
Full Text Search... Done
libgit2-dev/focal 0.28.4+dfsg.1-2 amd64
  low-level Git library (development files)
```

**Comment by bit-man** [comment]
```
$ mpm --verbosity DEBUG install libgit2-dev
debug: Verbosity set to DEBUG.
debug: Search for configuration in default location...
debug: No default configuration found.
debug: No configuration provided.
debug: mpm, version 4.13.0
debug: {'username': '-', 'guid': '1e7bc627e4b26f2ec9631bad0f0a8bc', 'hostname': '-', 'hostfqdn': '-', 'uname': {'system': 'Linux', 'node': '-', 'release': '5.13.0-39-generic', 'version': '#44~20.04.1-Ubuntu SMP Thu Mar 24 16:43:35 UTC 2022', 'machine': 'x86_64', 'processor': 'x86_64'}, 'linux_dist_name': '', 'linux_dist_version': '', 'cpu_count': 2, 'fs_encoding': 'utf-8', 'ulimit_soft': 1024, 'ulimit_hard': 1048576, 'cwd': '-', 'umask': '0o2', 'python': {'argv': '-', 'bin': '-', 'version': '3.8.10 (default, Mar 15 2022, 12:22:08) [GCC 9.4.0]', 'compiler': 'GCC 9.4.0', 'build_date': 'Mar 15 2022 12:22:08', 'version_info': [3, 8, 10, 'final', 0], 'features': {'openssl': 'OpenSSL 1.1.1f  31 Mar 2020', 'expat': 'expat_2.2.9', 'sqlite': '3.31.1', 'tkinter': '', 'zlib': '1.2.11', 'unicode_wide': True, 'readline': True, '64bit': True, 'ipv6': True, 'threading': True, 'urandom': True}}, 'time_utc': '2022-04-16 23:03:24.007584', 'time_utc_offset': -8.0, '_eco_version': '1.0.1'}
debug: apm CLI not found.
warning: Skip unavailable apm manager.
debug: apt CLI found at /usr/bin/apt
debug: ► /usr/bin/apt --version
debug:   apt 2.0.6 (amd64)
debug: Extracted version: '2.0.6'
debug: apt CLI found at /usr/bin/apt
debug: ► /usr/bin/apt version apt
error:   WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
error: 
error:   E: Invalid operation version
warning: Skip unavailable apt-mint manager.
debug: brew CLI not found.
warning: Skip unavailable brew manager.
debug: composer CLI not found.
warning: Skip unavailable composer manager.
debug: dnf CLI not found.
debug: yum CLI not found.
warning: Skip unavailable dnf manager.
debug: flatpak CLI not found.
warning: Skip unavailable flatpak manager.
debug: gem CLI not found.
warning: Skip unavailable gem manager.
debug: npm CLI not found.
warning: Skip unavailable npm manager.
debug: opkg CLI not found.
warning: Skip unavailable opkg manager.
debug: pacman CLI not found.
warning: Skip unavailable pacman manager.
debug: python3 CLI found at /usr/bin/python3
debug: ► /usr/bin/python3 -m pip --no-color --version
debug:   pip 20.0.2 from /usr/lib/python3/dist-packages/pip (python 3.8)
debug: Extracted version: '20.0.2'
debug: snap CLI found at /usr/bin/snap
debug: ► /usr/bin/snap --version
debug:   snap    2.54.3+20.04.1ubuntu0.2
debug:   snapd   2.54.3+20.04.1ubuntu0.2
debug:   series  16
debug:   ubuntu  20.04
debug:   kernel  5.13.0-39-generic
debug: Extracted version: '2.54.3+20.04.1ubuntu0.2'
debug: code CLI not found.
warning: Skip unavailable vscode manager.
debug: yarn CLI not found.
warning: Skip unavailable yarn manager.
Package manager order: 
debug: Try to install libgit2-dev with apt.
debug: ► /usr/bin/apt search ^libgit2-dev$ --quiet --names-only
debug:   Sorting...
debug:   Full Text Search...
debug:   libgit2-dev/focal 0.28.4+dfsg.1-2 amd64
debug:     low-level Git library (development files)
warning:   WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
warning: No libgit2-dev package found on apt.
debug: Try to install libgit2-dev with pip.
warning: No way to search for libgit2-dev with pip. Try to directly install it.
Install libgit2-dev package from pip...
debug: ► /usr/bin/python3 -m pip --no-color install libgit2-dev
error:   ERROR: Could not find a version that satisfies the requirement libgit2-dev (from versions: none)
error:   ERROR: No matching distribution found for libgit2-dev
warning: Could not install libgit2-dev with pip.
debug: Try to install libgit2-dev with snap.
debug: ► /usr/bin/snap --color=never find libgit2-dev
error:   error: unknown flag `color'
warning: No libgit2-dev package found on snap.

```

**Comment by bit-man** [comment]
This time the issue seems to be with apt search output parsing

```
debug: Try to install libgit2-dev with apt.
debug: ► /usr/bin/apt search ^libgit2-dev$ --quiet --names-only
debug:   Sorting...
debug:   Full Text Search...
debug:   libgit2-dev/focal 0.28.4+dfsg.1-2 amd64
debug:     low-level Git library (development files)
warning:   WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
warning: No libgit2-dev package found on apt.
```

**Comment by kdeldycke** [comment]
Thanks for the bug report @bit-man !

First, I just fixed that log in 06cb14bf0b74003c25fdfed7a66b4bf7ca843a79:
```
Package manager order: 
```

Now for the main bug, yes, it seems `apt search` output has evolved a bit. Give me some time so I can investigate deeper...

**Comment by kdeldycke** [comment]
```
debug: ► /usr/bin/snap --color=never find libgit2-dev
error:   error: unknown flag `color'
warning: No libgit2-dev package found on snap.
```

I just fixed this issue with b38ff81b02ea0bcd91e505ab7d5e8fc8b0f4203c. It was just a matter of wrong parameter order.

**Comment by kdeldycke** [comment]
OK, I think I properly fixed the issue by adding the missing calls to `sudo`. See: 656a4dab8b8eae15cdbacc64d12bb3d00491450c

**Comment by bit-man** [comment]
Shouldn't this bug remain open until main issue is fixed?
In 4.13.1 version still can't install and can't see it fixed in other commits either

**Comment by kdeldycke** [comment]
Yes, you're right @bit-man. The fix wasn't complete. I think I just fixed it for good in 15f122a8006e970ab81dcbf838c2929d8679a392.

This will be available in the upcoming v5.0.0 of `mpm`.
